### PR TITLE
fix(gallery): deduplicate slopegraph eyebrow number and strengthen sync verifier

### LINKED ISSUE #137: [Bug]: gallery eyebrow number 30 duplicated for Treemap and Slopegraph
### Area

A diagram example

### What happened?

In assets/index.html, both the Treemap tab (type 30) and the Slopegraph tab
were assigned eyebrow number 30. The duplicate is invisible at runtime — both
tabs load correctly — but breaks the sequential display index that users rely
on to scan the gallery.

Additionally, verify-docs-sync.py's check_gallery had no guard for duplicate
eyebrow numbers or for missing dark/full variants on non-single tabs, so the
regression passed all existing gates undetected.

### Steps to reproduce

1. Open skills/diagram-design/assets/index.html
2. Search for eyebrow">30
3. Observe two matches: Treemap and Slopegraph

### Environment

_No response_

### Validation gates

- [ ] `python3 scripts/test-lint-a11y.py`
- [ ] `python3 scripts/lint-skin.py --all --baseline`
- [ ] `python3 scripts/verify-drawio-import.py`
- [ ] `python3 scripts/verify-mermaid-import.py`
- [ ] I haven't run any gates

### Relevant output

```text
grep 'eyebrow">30' skills/diagram-design/assets/index.html
# <span class="eyebrow">30</span>Treemap
# <span class="eyebrow">30</span>Line · slopegraph
```

### PR DESCRIPTION
**Pull request by Rupam0710**
## What and why

The gallery (`assets/index.html`) assigns a short eyebrow number to each type
tab as a visual index. When Slopegraph was added it was accidentally given `30`,
the same number already assigned to Treemap. The duplicate was invisible at
runtime (both tabs still loaded their files correctly) but broke the sequential
display index users rely on to scan the gallery.

Additionally, `verify-docs-sync.py`'s `check_gallery` function only validated
that the light variant of each tab exists on disk. It had no check for duplicate
eyebrow numbers — which is why this slipped through — and no check that
non-single tabs have their dark and full variants.

### `skills/diagram-design/assets/index.html`
- Corrected slopegraph eyebrow `30` → `31` (Treemap retains `30`).

### `scripts/verify-docs-sync.py`
Extended `check_gallery` with two new guards:

1. **Duplicate eyebrow detection** — errors if any two tab buttons share the
   same display number. This would have caught the bug before it shipped.
2. **Full variant coverage for non-single tabs** — errors if a tab without
   `data-single` is missing its `-dark` or `-full` example file on disk,
   closing a gap where an incomplete type addition would pass all existing gates.

## Validation

All existing gates pass:

Closes #137

### CODE REVIEW

File: scripts/verify-docs-sync.py [hunk 1]
```diff
@@ -146,6 +146,36 @@ def check_gallery(errors: list[str]) -> None:
     for name in sorted(types):
         if f"example-{name}.html" not in on_disk:
             errors.append(f"gallery tab {name!r} points at a missing example-{name}.html")
+    # Detect duplicate eyebrow numbers (display labels on each tab button).
+    seen_eyebrows: dict[str, int] = {}
```
[the remark this answers is not in the document]
Comment (Rupam0710) [reply scripts/verify-docs-sync.py]: Done — six focused cases added to test-verify-docs-sync.py in commit `54f2bed`:

- duplicate eyebrow number is caught
- unique eyebrow numbers produce no error
- non-single tab missing `-dark` variant is caught
- non-single tab missing `-full` variant is caught
- `data-single` tab with only the light file passes (no false positive)
- complete non-single tab (all three variants present) passes

All six pass locally. The `data-single` false-positive path is explicitly covered as its own case.

**Review by philyuchkoff — COMMENTED** [review]
# Review: fix(gallery) deduplicate slopegraph eyebrow number — PR #136

@Rupam0710 

Clean fix that closes #137. The problem is simple — Treemap and Slopegraph both had eyebrow `30` — but the PR goes further than just bumping the eyebrow number: it adds duplicate eyebrow detection, variant-file existence checks, `data-single` support, and full adversarial test coverage.

**Code review**

1. **Eyebrow number fix** (`index.html:299`): `30` → `31` for slopegraph. Correct — Slopegraph is type #31 (one position after Treemap).

2. **`verify-docs-sync.py` additions:**

   a. **Duplicate eyebrow detection** (lines 150–161): `seen_eyebrows` dict catches duplicate `<span class="eyebrow">NUM</span>`. Simple and correct. One nit: `m.start()` is stored in the dict but never read — dead store, non-blocking.

   b. **`data-single` parsing** (lines 162–169): iterates `<button[^>]+>` tags, collects types with `data-single`. These tabs don't require `-dark`/`-full` variants. Fine in practice — `data-type`/`data-single` only appear on gallery tab buttons, not on generic buttons.

   c. **Variant existence checks** (lines 173–181): after `on_disk` is populated, non-single tabs now verify that `-dark` and `-full` variant files exist on disk. Right place, right check — catches genuine CI gaps early.

3. **Test coverage** (`test-verify-docs-sync.py`): six new test cases covering duplicate eyebrows, unique eyebrows (false-positive guard), missing `-dark`, missing `-full`, `data-single` exemption, and complete tab pass. Well-isolated via monkey programming of `verify.GALLERY` and `verify.ASSET_DIR` within temporary directories — no side effects.

**Version bump**: 2.6.1 → 2.6.2 correct.

LGTM, approved.

**Comment by Rupam0710** [comment]
Thanks for the nit, @philyuchkoff — good spot. Cleaned it up to a `set[str]` and pushed.

**Review by cathrynlavery — CHANGES_REQUESTED** [review]
The duplicate report is valid, but the proposed `31` and global uniqueness rule conflict with the gallery's established parent/variant model: Line/ridgeline share 20 and Scatter/bubble share 22. Slopegraph is also a Line variant and should use 20.

Please encode the relationship explicitly (for example, a `data-parent-type` binding), require a variant to reuse its declared parent's number, and enforce uniqueness only among independent parent types. Add tests for allowed parent/variant reuse, a variant with the wrong number, an unrelated duplicate, and a missing parent. Then rebase, choose a unique synchronized version, and rerun the full current suite.

**Comment by Rupam0710** [comment]
Hi @cathrynlavery — addressed your feedback, here's what changed:

- Reverted slopegraph eyebrow `31` → `20` (Line variant, shares its parent's number)
- Added `data-parent-type="line"` to `slopegraph` and `ridgeline`, and `data-parent-type="scatter"` to `bubble`
- Updated verify-docs-sync.py to enforce uniqueness only among independent types, require variants to match their parent's eyebrow exactly, and error on a missing parent
- Added three new test cases: allowed parent/variant reuse, wrong variant number, missing parent — all pass

Also merged latest upstream/main and bumped manifests to `2.6.7`. Could you please take another look? Thanks!

**Comment by cathrynlavery** [comment]
merged, thanks for contributing!
